ext/hash/config.m4

```diff
@@ -17,31 +26,55 @@ else
   AC_MSG_CHECKING([if we're at 64-bit platform])
   AS_IF([test "$ac_cv_sizeof_long" -eq 4],[
     AC_MSG_RESULT([no])
+    if test $ac_cv_target_x86 = yes; then 
+      dnl there are some early Pentium4 32bit CPUs with SSE2 support, but don't waste effort optimizing musem-CPUs
+      dnl (the majority of 32bit CPUs does not have SSE2)
+      PHP_HASH_CFLAGS="$PHP_HASH_CFLAGS -DBLAKE3_NO_SSE2 -DBLAKE3_NO_SSE41 -DBLAKE3_NO_AVX2 -DBLAKE3_NO_AVX512"
+    fi
     SHA3_DIR="sha3/generic32lc"
     SHA3_OPT_SRC="$SHA3_DIR/KeccakP-1600-inplace32BI.c"
   ],[
     AC_MSG_RESULT([yes])
+    if test $ac_cv_target_x86 = yes; then 
+      if test $ac_cv_target_windows = yes; then
+        dnl x86_64 windows gnuc
+        EXT_HASH_BLAKE3_SOURCES="$EXT_HASH_BLAKE3_SOURCES blake3/upstream_blake3/c/blake3_avx512_x86-64_windows_gnu.S blake3/upstream_blake3/c/blake3_avx2_x86-64_windows_gnu.S blake3/upstream_blake3/c/blake3_sse41_x86-64_windows_gnu.S blake3/upstream_blake3/c/blake3_sse2_x86-64_windows_gnu.S"
+      else
+        if test $ac_cv_target_unix = yes; then 
+          dnl x86_64 unix gnuc
+          EXT_HASH_BLAKE3_SOURCES="$EXT_HASH_BLAKE3_SOURCES blake3/upstream_blake3/c/blake3_avx512_x86-64_unix.S blake3/upstream_blake3/c/blake3_avx2_x86-64_unix.S blake3/upstream_blake3/c/blake3_sse41_x86-64_unix.S blake3/upstream_blake3/c/blake3_sse2_x86-64_unix.S"
+        else
+          dnl blake still has C instrictics versions (eg ext/hash/blake3/upstream_blake3/c/blake3_avx512.c  ) which COULD be used in this situation, 
+          dnl we're targeting x86_64 but neither windows nor unix.. 
+          dnl i have nothing to test this on, so i'm just using portable for now, feel free to fix it.
+          PHP_HASH_CFLAGS="$PHP_HASH_CFLAGS -DBLAKE3_NO_SSE2 -DBLAKE3_NO_SSE41 -DBLAKE3_NO_AVX2 -DBLAKE3_NO_AVX512"
+        fi
+      fi
+    fi
```

Comment:
```suggestion
    AS_CASE([$host_cpu],[x86_64*|amd64*], [
      if test $ac_cv_target_windows = yes; then
        dnl x86_64 windows gnuc
        EXT_HASH_BLAKE3_SOURCES="$EXT_HASH_BLAKE3_SOURCES blake3/upstream_blake3/c/blake3_avx512_x86-64_windows_gnu.S blake3/upstream_blake3/c/blake3_avx2_x86-64_windows_gnu.S blake3/upstream_blake3/c/blake3_sse41_x86-64_windows_gnu.S blake3/upstream_blake3/c/blake3_sse2_x86-64_windows_gnu.S"
      else
        if test $ac_cv_target_unix = yes; then 
          dnl x86_64 unix gnuc
          EXT_HASH_BLAKE3_SOURCES="$EXT_HASH_BLAKE3_SOURCES blake3/upstream_blake3/c/blake3_avx512_x86-64_unix.S blake3/upstream_blake3/c/blake3_avx2_x86-64_unix.S blake3/upstream_blake3/c/blake3_sse41_x86-64_unix.S blake3/upstream_blake3/c/blake3_sse2_x86-64_unix.S"
        else
          dnl blake still has C instrictics versions (eg ext/hash/blake3/upstream_blake3/c/blake3_avx512.c  ) which COULD be used in this situation, 
          dnl we're targeting x86_64 but neither windows nor unix.. 
          dnl i have nothing to test this on, so i'm just using portable for now, feel free to fix it.
          PHP_HASH_CFLAGS="$PHP_HASH_CFLAGS -DBLAKE3_NO_SSE2 -DBLAKE3_NO_SSE41 -DBLAKE3_NO_AVX2 -DBLAKE3_NO_AVX512"
        fi
      fi
    ])
```

This detects the x86_64 processor. x86_64 is detected as word amd64 on FreeBSD. The $host_os is in Autoconf used for the target CPU when cross-compiling or as a host CPU when building on the same machine as the program will be also run.